pkg/function/service.go

```diff
@@ -293,3 +333,12 @@ func getValueFromMetadata(md grpcmd.MD, k string) (string, error) {
 	}
 	return value, nil
 }
+
+func isClosed(c chan Message) bool {
+	select {
+	case <-c:
```

Comment:
This can only be used when we know there would be any more data send to the channel, which is exactly the case we are facing. Using this article from go 101 as the reference https://go101.org/article/channel-closing.html.

Reply:
👍 good read..

Question - if reading `messageCh` is slow in line 159-170, which means line 154 might be executed earlier, will that be a problem?

Reply:
Good question, tried this in playground and a slow receiver still can get all data https://go.dev/play/p/oBIyPVmbA4p

Reply:
No, it missed 6 :)

Reply:
Because of the slow consumer, the `isClosed()` function used one message.

Reply:
hmmmm...actually, you are right.

Reply:
This is interesting, a channel can not be read the data written in the same goroutine...

Reply:
Right, exactly.